src/stdlib/SDL_malloc.c

```diff
@@ -2531,11 +2531,14 @@ static int init_mparams(void) {
       int fd;
       unsigned char buf[sizeof(size_t)];
       /* Try to use /dev/urandom, else fall back on using time */
-      if ((fd = open("/dev/urandom", O_RDONLY)) >= 0 &&
-          read(fd, buf, sizeof(buf)) == sizeof(buf)) {
-        s = *((size_t *) buf);
+      if (fd = open("/dev/urandom", O_RDONLY) < 0) {
```

Comment:
Please keep the parentheses around assignment to fd:
`    if ((fd = open("/dev/urandom", O_RDONLY)) < 0) {`